### PR TITLE
Basic JSDoc + quick-and-easy testing setup

### PR DESCRIPTION
**Pull request by kungfooman**
Basic setup for getting types checked with quick and easy testing:

image

Now we can start enjoying type checking like this:

image

:wink: :wine_glass:

### CODE REVIEW

File: devtools.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<script type="module">
+    import * as aimless from "./src/aimless.js";
+    import * as utils from "./src/utils.js";
+    Object.assign(window, { ...aimless, ...utils });
+</script>
+<pre>
+// just type any code in F12/DevTools like this:
+floatRange(10, 20); // e.g. 17.68330329779974
+defaultEngine(); // e.g. 0.7867804299919436
+</pre>
```
Comment (ChrisCavs) [inline devtools.html:10]: there's a similar file for manual testing here: https://github.com/ChrisCavs/aimless.js/blob/main/test/manual.html

File: src/int-range.js [hunk 2]
```diff
@@ -2,9 +2,9 @@ import { defaultEngine, generateCurry, randIntRange } from './utils.js'
 
 /**
  * Returns a random number within the integer range
- * @param min Minimum integer
- * @param max Maximum integer
- * @param engine PRNG of choice
+ * @param {number} min Minimum integer
```
Comment (ChrisCavs) [inline src/int-range.js:5]: I'm unfamiliar with jsDoc, but I found the documentation here: https://jsdoc.app/tags-type.html

I'm curious, does this play well with TypeScript applications?

Comment (kungfooman) [reply src/int-range.js:5]: Yep, I just want to push you a bit into the right direction here, once you did the rest of JSDoc's, you can generate `aimless.d.ts` (which is what goes later into the npm package)

I just added the command for that already, if you want to see:

image

File: devtools.html (hunk 1 above)
Comment (kungfooman) [reply devtools.html:10]: Thank you, didn't see that one. It just has three problems rendering it less useful for testing:
- no access to all exported functions:
image
- requires a build step, which slows down testing and debugging
- requires to prefix everything with `aimless`, deviating from modern syntax or how you would write code in any normal setting (either CJS or MJS)

File: src/int-range.js (hunk 2 above)
Comment (ChrisCavs) [reply src/int-range.js:5]: i see, so i need typescript as a dev dependency.  once I run the command and generate the types into the types folder, will they automatically be bundled into `dist` on build?

Comment (ChrisCavs) [reply src/int-range.js:5]: i have the type files generated, and i added a rollup step to bundle them, but it's not working.  not sure how to get these types declarations actually in the dist folder

**Pull request closed without merging**